testing/src/test/java/org/motechproject/nms/testing/it/imi/ImiSMSNotificationServiceBundleIT.java

```diff
@@ -0,0 +1,84 @@
+package org.motechproject.nms.testing.it.imi;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.motechproject.alerts.contract.AlertsDataService;
+import org.motechproject.alerts.domain.Alert;
+import org.motechproject.nms.imi.service.SettingsService;
+import org.motechproject.nms.imi.service.SmsNotificationService;
+import org.motechproject.nms.testing.service.TestingService;
+import org.motechproject.server.config.SettingsFacade;
+import org.motechproject.testing.osgi.BasePaxIT;
+import org.motechproject.testing.osgi.container.MotechNativeTestContainerFactory;
+import org.motechproject.testing.utils.TestContext;
+import org.ops4j.pax.exam.ExamFactory;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerSuite;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+@ExamFactory(MotechNativeTestContainerFactory.class)
+public class ImiSMSNotificationServiceBundleIT extends BasePaxIT {
+
+    private static final String SMS_NOTIFICATION_URL = "imi.sms.notification.url";
+
+    private static final String MAX_NOTIFICATION_RETRY_COUNT = "imi.notification_retry_count";
+
+    @Inject
+    SettingsService settingsService;
+
+    @Inject
+    SmsNotificationService smsNotificationService;
+
+    @Inject
+    AlertsDataService alertsDataService;
+
+    @Inject
+    TestingService testingService;
+
+    SettingsFacade settingsFacade;
+
+    String oldEndpoint, oldRetryCount;
+
+    @Before
+    public void cleanupDatabase() {
+        testingService.clearDatabase();
+
+        settingsFacade = settingsService.getSettingsFacade();
+        oldEndpoint = settingsFacade.getProperty(SMS_NOTIFICATION_URL);
+        oldRetryCount = settingsFacade
+                .getProperty(MAX_NOTIFICATION_RETRY_COUNT);
+    }
+
+    @After
+    public void restore() {
+        settingsFacade.setProperty(SMS_NOTIFICATION_URL, oldEndpoint);
+        settingsFacade.setProperty(MAX_NOTIFICATION_RETRY_COUNT, oldRetryCount);
+    }
+
+    @Test
```

Comment:
@koshalt  : We are trying to check the code behavior upon receiving 202 from IMI. This is the only relation of this test case with #434 .
I will add some description as well.